src/Result/Invoice.php

```diff
@@ -24,6 +24,51 @@ class Invoice extends AbstractResult
 
     public const ADDITIONAL_STATUS_PAID_LATE = 'PaidLate';
 
+    public function getId(): string
+    {
+        return $this->getData()['amount'];
+    }
+
+    public function getAmount(): float
+    {
+        return $this->getData()['amount'];
+    }
+
+    public function getCurrency(): string
+    {
+        return $this->getData()['currency'];
+    }
+
+    public function getType(): string
+    {
+        return $this->getData()['type'];
+    }
+
+    public function getCheckoutLink(): string
+    {
+        return $this->getData()['checkoutLink'];
+    }
+
+    public function getCreatedTime(): int
+    {
+        return $this->getData()['createdTime'];
+    }
+
+    public function getExpirationTime(): int
+    {
+        return $this->getData()['expirationTime'];
+    }
+
+    public function getMonitoringTime(): int
+    {
+        return $this->getData()['monitoringTime'];
+    }
+
+    public function isArchived(): bool
+    {
+        return $this->getData()['archived'];
+    }
+
     public function isPaid(): bool
```

Comment:
Not sure if I like this name. Maybe rename to hasPayment()?

What do you think @ndeet ?
We have discussed a similar issue before.

Reply:
Yes I ran into that with integrating OpenCart, we can't rename but will deprecate I think. Out of scope for this PR though.